lib/statsd.js

```diff
@@ -38,20 +51,40 @@ const Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
     let socket;
     let errMessage;
 
-    if (args.protocol === 'tcp') {
+    if (args.protocol === TCP) {
       try {
         socket = net.connect(args.port, args.host);
         socket.setKeepAlive(true);
-      } catch (e) {
-        errMessage = `Could not establish connection to ${args.host}:${args.port}`;
-        if (instance.errorHandler) {
-          instance.errorHandler(new Error(errMessage));
-        } else {
-          console.log(errMessage);
+      } catch (err) {
+        errMessage = `Could not establish connection to ${args.host}:${args.port}: ${err}`;
+      }
+    } else if (args.protocol === UDS) {
+      if (! unixDgram) {
+        errMessage = 'The library unix_dgram, needed for UDS, is not available on your OS or version of Node.';
+      }
+      else {
+        const udsPath = args.path ? args.path : '/var/run/datadog/dsd.socket';
```

Comment:
I like how you defined the protocol constants! Maybe defaults for options including the datadog uds path can also be assigned to constants

Reply:
Great point.  Just adding this one for now.